### PR TITLE
Identify reproducible test issues

### PR DESCRIPTION
**Pull request by perlpunk**
* Identify reproducible test issues
* Add link to documentation

Issue: https://progress.opensuse.org/issues/132272

Screenshot:
investigate-comment-linkdoc

### CODE REVIEW

File: openqa-investigate [hunk 1]
```diff
@@ -260,10 +261,13 @@ post-investigate() {
...
         comment+=" failed, likely not a sporadic failure"
         local product_issue=false
-        is-product-issue "$origin_job_id" || return $?
+        local test_issue=false
+        identify-issue-type "$origin_job_id" || return $?
 
         if "$product_issue"; then
             comment+="."$'\n'"Jobs including the last good build are ok, likely a product issue"
+        elif "$test_issue"; then
+            comment+="."$'\n'"Jobs including the last good build are ok, likely a test issue"
```
Comment (okurz) [inline openqa-investigate:270]:
```suggestion
            comment+="."$'\n'"Jobs including the last good test are ok, likely a test issue"
```

File: test/02-investigate.t [hunk 2]
```diff
@@ -178,6 +194,14 @@ test-post-investigate() {
     is "$rc" 0 'early return (other test in cluster) (30005)'
     has "$got" "is not a retry" "vim-other vs. vim (30005)"
 
+    # test issue
+    try investigate 30031
+    is "$rc" 2 'mocked function returned failure (30031)'
+    has "$got" "Commenting 3003" "Posting comment on OPENQA_INVESTIGATE_ORIGIN (30031)"
+    has "$got" "Investigate retry job **vim:investigate:retry**" "retry test name appears in comment(30031)"
+    has "$got" "likely not a sporadic" "not sporadic (30031)"
+    has "$got" "likely a test issue" "product issue (30031)"
```
Comment (okurz) [inline test/02-investigate.t:203]: I assume the test description string should reference test issue, not product issue

File: README.md [hunk 3]
```diff
@@ -143,6 +143,41 @@ following scripts are provided:
...
+in the test code, it will trigger the test using the git sha from that last good
+test, e.g. `CASEDIR=orginal_git_url#abc123`.
+
+The `vim:investigate:last_good_build:TW.314` is a retry using the last good
+build.  It searches for the last good test, and if it has a different build
+than the current failing test, it triggers the original test with that build,
+e.g.  `BUILD=TW.314`.
+
+The `vim:investigate:last_good_tests_and_build:abc123+TW.314` is a combination
+of the other two.
```
Comment (grisu48) [inline README.md:173]:
```suggestion
of the other two, i.e. a retry of the last good build (`BUILD=TW.314`) with the last good test code (`CASEDIR=orginal_git_url#abc123`).
```
I would be also explicitly state the example settings.

File: README.md [hunk 4]
```diff
@@ -143,6 +143,41 @@ following scripts are provided:
   found, "openqa-investigate". Compare to section
   "Combine auto-review and openqa-investigate"
 
+### openqa-investigate comments
```
Comment (okurz) [inline README.md:146]: As there is already the section in https://github.com/os-autoinst/scripts/pull/265/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R94 I would make that

```suggestion
### More details and examples about openqa-investigate comments
```

**Review by grisu48 — COMMENTED** [review]
Thank you Tina! One small nitpicking suggestion but I find this already useful now.

Not sure if possible, but once this is merged it would be super helpful if there would be a direct link to this documentation directly in the investigate job comment in openQA.

File: README.md [hunk 5]
```diff
@@ -143,6 +143,41 @@ following scripts are provided:
...
+
+The `vim:investigate:last_good_build:TW.314` is a retry using the last good
+build.  It searches for the last good test, and if it has a different build
+than the current failing test, it triggers the original test with that build,
+e.g.  `BUILD=TW.314`.
+
+The `vim:investigate:last_good_tests_and_build:abc123+TW.314` is a combination
+of the other two.
+
+With the results of all those tests it can be seen quicly whether something is
```
Comment (okurz) [inline README.md:175]:
```suggestion
With the results of all those tests it can be seen quickly whether something is
```

**Review by okurz — CHANGES_REQUESTED** [review]
Very nice addition!

File: openqa-investigate (hunk 1 above)
Comment (perlpunk) [reply openqa-investigate]: done

File: README.md [hunk 6]
```diff
@@ -143,6 +143,43 @@ following scripts are provided:
...
+The first job `vim:investigate:retry` is a simple retry with the same settings,
+in order to find out if the failure is reproducible consistently.
+
+The `vim:investigate:last_good_tests:abc123` is a retry which uses the last
+good test code. It searches for the last good test, and if there were changes
+in the test code, it will trigger the test using the git sha from that last good
+test, e.g. `CASEDIR=orginal_git_url#abc123`.
+
+The `vim:investigate:last_good_build:TW.314` is a retry using the last good
+build.  It searches for the last good test, and if it has a different build
```
Comment (Martchus) [inline README.md:168]:
```suggestion
build. It searches for the last good test, and if it has a different build
```

File: README.md [hunk 7]
```diff
@@ -143,6 +143,43 @@ following scripts are provided:
...
+than the current failing test, it triggers the original test with that build,
+e.g.  `BUILD=TW.314`.
```
Comment (Martchus) [inline README.md:170]:
```suggestion
e.g. `BUILD=TW.314`.
```

File: README.md [hunk 8]
```diff
@@ -143,6 +143,43 @@ following scripts are provided:
...
+
+The `vim:investigate:last_good_tests_and_build:abc123+TW.314` is a combination
+of the other two, i.e. a retry of the last good build (`BUILD=TW.314`) with the
+last good test code (`CASEDIR=orginal_git_url#abc123`).
+
+With the results of all those tests it can be seen quickly whether something is
+an issue with the new build or the changes in the test code or something else.
+
+As an example, If the `retry` fails, `last_good_tests` fails, but
```
Comment (Martchus) [inline README.md:179]:
```suggestion
As an example, if the `retry` fails, `last_good_tests` fails, but
```

File: README.md (hunk 6 above)
Comment (perlpunk) [reply README.md:168]: I'm using the vim `gq` mapping to wrap blocks, it inserts those spaces. They will be collapsed to one space anyway. Do I need to manually delete them?
Maybe I'm using `gq` wrongly?

**Comment by perlpunk** [comment]
I tested it locally and added a screenshot

File: README.md (hunk 6 above)
Comment (okurz) [reply README.md]: Well, I have the same experience and I am deleting them manually by searching for double spaces if they don't jump to my eye already in before :)

**Review by okurz — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

**Pull request merged**